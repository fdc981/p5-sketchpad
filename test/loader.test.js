/**
 * @jest-environment node
 */
import compiler from './compiler.js';

import path from 'path';
import webpack from 'webpack';
import { createFsFromVolume, Volume } from 'memfs';
import { getModuleSource, getWarnings, getErrors } from './helpers';


test('converts a simple global-mode p5.js sketch', async () => {
  const stats = await compiler('./fixtures/example.p5.js');

  expect(getModuleSource(stats)).toMatchSnapshot("result");
  expect(getWarnings(stats)).toHaveLength(0);
  expect(getErrors(stats)).toHaveLength(0);
});

test('converts a global-mode sketch containing a function definition', async () => {
  const stats = await compiler('./fixtures/example-function.p5.js');

  expect(getModuleSource(stats)).toMatchSnapshot("result");
  expect(getWarnings(stats)).toHaveLength(0);
  expect(getErrors(stats)).toHaveLength(0);
});

test('converts a global-mode sketch containing a class', async () => {
  const stats = await compiler('./fixtures/example-class.p5.js');

  expect(getModuleSource(stats)).toMatchSnapshot("result");
  expect(getWarnings(stats)).toHaveLength(0);
  expect(getErrors(stats)).toHaveLength(0);
});
