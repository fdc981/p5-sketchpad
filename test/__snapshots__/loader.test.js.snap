// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts a global-mode sketch containing a function definition: result 1`] = `
"__webpack_nonce__ = 'c29tZSBjb29sIHN0cmluZyB3aWxsIHBvcCB1cCAxMjM=';
  import p5 from 'p5';

  let sketch = function($_p) {
    /**
 * Draws a polygon centered at x, y, with given radius and number of sides
 * numSides.
 */
function polygon(x, y, radius, numSides) {
  $_p.beginShape();
  for (let i = 0; i < numSides; i++) {
    $_p.vertex(x + radius * Math.cos(2 * Math.PI / numSides * i),
           y + radius * Math.sin(2 * Math.PI / numSides * i));
  }
  $_p.endShape($_p.CLOSE);
}

$_p.setup = function() {
  $_p.createCanvas(200, 200);
  $_p.noLoop();
}

$_p.draw = function() {
  polygon(100, 100, 40, 5);
}

  }

  let myp5 = new p5(sketch);
  "
`;

exports[`converts a simple global-mode p5.js sketch: result 1`] = `
"__webpack_nonce__ = 'c29tZSBjb29sIHN0cmluZyB3aWxsIHBvcCB1cCAxMjM=';
  import p5 from 'p5';

  let sketch = function($_p) {
    $_p.setup = function() {
  $_p.createCanvas(200, 200);
  $_p.noLoop();
}

$_p.draw = function() {
  $_p.stroke("black");
  $_p.fill("black");
  $_p.circle(50, 50, 50);
}

  }

  let myp5 = new p5(sketch);
  "
`;
